src/layer/boundary.js

```diff
@@ -242,6 +243,72 @@ export const country = {
   "source-layer": "boundary",
 };
 
+/**
+ * Returns an expression that converts the given country code to a
+ * human-readable name in the user's preferred language.
+ *
+ * @param code An expression that evaluates to an ISO 3166-1 alpha-3 country
+ *  code.
+ */
+function getCountryName(code) {
+  return [
+    "let",
+    "code",
+    code,
+    "countryNamesByCode",
+    ["literal", Label.countryNamesByCode],
+    [
+      "coalesce",
+      ["get", ["var", "code"], ["var", "countryNamesByCode"]],
+      // Fall back to the country code in parentheses.
+      ["concat", "(", ["var", "code"], ")"],
+    ],
+  ];
+}
+
+export const countryLabelLeft = {
+  id: "boundary_country_label_left",
+  type: "symbol",
+  paint: {
+    "text-color": {
+      base: 1.2,
+      stops: [
+        [3, `hsl(${Color.hueBorder}, 2%, 24%)`],
+        [7, `hsl(${Color.hueBorder}, 2%, 18%)`],
+      ],
+    },
+  },
+  layout: {
+    "symbol-placement": "line",
+    "text-font": ["OpenHistorical Bold"],
+    "text-size": {
+      stops: [
+        [3, 6],
+        [7, 10],
+      ],
+    },
+    "text-field": getCountryName(["get", "adm0_l"]),
+    "text-offset": [0, -1],
```

Comment:
Setting this offset causes glyphs to bunch up and spread out awkwardly at any bend in the geometry:

[<img src="https://user-images.githubusercontent.com/1231218/216784929-e32b0930-2a44-42a1-ae17-25de28c26a28.png" width="400" alt="Mackinac">](https://zelonewolf.github.io/openstreetmap-americana/#map=13/46.50721/-84.34302&language=en)

This is due to an upstream issue: mapbox/mapbox-gl-js#3718. It affects any label offset away from the geometry, such as road name labels: #514. I’ve tried to mitigate the bunching somewhat by applying a 0.1&nbsp;em letter spacing and limiting `text-max-angle` to 30&nbsp;degrees, but pushing these settings further would start to have visual tradeoffs or reduce the density of these labels.

Reply:
maplibre/maplibre-gl-js#2170